open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -80,123 +79,146 @@ public static IdempotencyValidationResult validateIdempotency(String idempotency
                     for (String consentId : consentIds) {
                         DetailedConsentResource consentRequest = consentCoreService.getDetailedConsent(consentId);
                         if (consentRequest != null) {
-                            // Compare the client ID sent in the request and client id retrieved from the database
-                            // to validate whether the request is received from the same client
-                            if (isClientIdsMatching(clientId, consentRequest.getClientID())) {
-                                // Check whether difference between two dates is less than the configured time
-                                if (isRequestReceivedWithinAllowedTime(consentRequest.getCreatedTime())) {
-                                    // Compare whether JSON payloads are equal
-                                    if (isJSONPayloadSimilar(consentRequest.getReceipt(), request)) {
-                                        log.debug("Payloads are similar and request received within allowed time." +
-                                                " Hence this is a valid idempotent request");
-                                        return new IdempotencyValidationResult(true, true,
-                                                consentRequest, consentId);
-                                    } else {
-                                        log.debug(IdempotencyConstants.ERROR_PAYLOAD_NOT_SIMILAR);
-                                        throw new ConsentManagementException(IdempotencyConstants
-                                                .ERROR_PAYLOAD_NOT_SIMILAR);
-                                    }
-                                } else {
-                                    log.debug(IdempotencyConstants.ERROR_AFTER_ALLOWED_TIME);
-                                    throw new ConsentManagementException(IdempotencyConstants
-                                            .ERROR_AFTER_ALLOWED_TIME);
-                                }
-                            } else {
-                                log.debug(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
-                                throw new ConsentManagementException(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
-                            }
+                            return validateIdempotencyConditions(consentManageData, consentRequest);
                         } else {
-                            log.debug(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
-                            throw new ConsentManagementException(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
+                            log.error(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
```

Comment:
Can't we log the consentId here?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661